UnitTests/Onboarding/ContextualOnboarding/ContextualOnboardingStateMachineTests.swift

```diff
@@ -684,6 +684,22 @@ class ContextualOnboardingStateMachineTests: XCTestCase {
         // Then
         XCTAssertEqual(stateMachine.state, .onboardingCompleted)
     }
+
+    @MainActor
+    func test_lastVisitTabDoesNotCreateRetainCycle() {
```

Comment:
This seems to fail on CI. Probably a timing issue. Maybe to avoid timing issue we can flip the test? What I mean is if we have a weak Tab, call stateMachine.updateStateFor(tab: tab!) and the nil the stateMachine then the tab should be nil too. Just an idea.